origins_generators/sources/delimited.py

```diff
@@ -75,47 +84,35 @@ def setup(self):
         if not self.options.columns:
             self.options.columns = _header
 
-    def parse_file(self):
-        uri = self.options.uri
-
-        name = os.path.splitext(os.path.basename(uri))[0]
-
-        if not utils.is_remote(self.options.uri):
-            uri = os.path.abspath(uri)
-
-        return {
-            'origins:id': name,
-            'prov:label': utils.prettify_name(name),
-            'prov:type': 'File',
-            'uri': uri,
-        }
-
     def parse_columns(self):
         columns = []
-
         for i, name in enumerate(self.options.columns):
-            column = {
-                'origins:id': name,
-                'prov:label': name,
-                'prov:type': 'Column',
-                'index': i,
-            }
-
-            columns.append(column)
-
+            columns.append(name)
         return columns
 
     def parse(self):
-        file = self.parse_file()
-        self.document.add('entity', file)
-
-        columns = self.parse_columns()
-
-        for column in columns:
-            self.document.add('entity', column)
-
-            self.document.add('wasInfluencedBy', {
-                'prov:influencer': file,
-                'prov:influencee': column,
-                'prov:type': 'origins:Edge',
-            })
+        FIELDS = self.parse_columns()
+        with open(self.options.uri, 'rU', newline='') as f:
+            yield [
+                'operation',
+                'domain',
+                'entity',
+                'attribute',
+                'value',
+                'valid_time'
+            ]
+
+            reader = csv.DictReader(f, fieldnames=FIELDS,
+                                    delimiter=self.options.delimiter)
+            next(reader)
+            for line in reader:
+                for key, value in line.items():
+                    if key != FIELDS[0]:
```

Comment:
Sorry, I am confusing myself. The delimited generator **only** needs access to the columns, the first line of the file which is [this](https://github.com/swanijam/origins-generator-service/blob/master/origins_generators/sources/delimited.py#L69-L85) already does. So this `parse` method only needs to iterate over the columns and yield the names:

```python
def parse(self):
    # Header for facts
    yield (
        'operation',
        'domain',
        'entity',
        'attribute_domain',
        'attribute',
        'value',
        'valid_time',
    )

    for i, field in self.options.columns:
        # Assert entity ident which behaves like an identity function
        # See: https://en.wikipedia.org/wiki/Identity_function
        yield (
            'assert',
            self.options.domain,
            field,
            'origins.attrs',
            'ident',
            field,
            self.options.time,
        )
```